control-plane-operator/hostedclusterconfigoperator/controllers/resources/registry/admissionpolicies.go

```diff
@@ -48,15 +49,44 @@ func ReconcileRegistryConfigValidatingAdmissionPolicies(ctx context.Context, hcp
 	return nil
 }
 
+// reconcileRegistryConfigManagementStateValidatingAdmissionPolicy reconciles the Validating Admission Policy
+// that controls access to the Image Registry config managementState field.
+//
+// During normal operation, it prevents users from setting managementState to 'Removed' except for the HCCO user.
+// During cluster deletion, it deletes the policy and its binding to allow HCCO cleanup operations.
 func reconcileRegistryConfigManagementStateValidatingAdmissionPolicy(ctx context.Context, hcp *hyperv1.HostedControlPlane, client client.Client, createOrUpdate upsert.CreateOrUpdateFN) error {
+	log := ctrl.LoggerFrom(ctx)
 	registryConfigManagementStateAdmissionPolicy := AdmissionPolicy{Name: AdmissionPolicyNameManagementState}
 	registryConfigManagementStateAPIVersion := []string{imageregistryv1.GroupVersion.Version}
 	registryConfigManagementStateAPIGroup := []string{imageregistryv1.GroupVersion.Group}
 	registryConfigManagementStateResources := []string{
 		"configs",
 	}
 
-	denyRemovedManagementStateValidation.Expression = "object.spec.managementState != 'Removed' || request.userInfo.username == 'system:hosted-cluster-config'"
+	if !hcp.DeletionTimestamp.IsZero() {
+		// During cluster deletion, delete the admission policy and its binding to allow HCCO cleanup
```

Comment:
```suggestion
		// During cluster deletion, delete the admission policy and its binding to allow CIRO cleanup
```
maybe also add more context on why CIRO need to do that